lib/include/domain_participant/builtin/discovery_config/DiscoveryConfig.hpp

```diff
@@ -0,0 +1,1057 @@
+ * @param[in] count Builtin discovery number of initial announcements.
+ *
+ * @throw ElementInvalid Exception if the provided number of initial announcements is not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_initial_announcements_count(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& count);
+
+/**
+ * @brief Set the Domain Participant builtin discovery initial announcements duration.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] period_sec Builtin discovery initial announcements duration seconds.
+ * @param[in] period_nanosec Builtin discovery initial announcements duration nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided initial announcements duration arguments are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_initial_announcements_period(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& period_sec,
+        const std::string& period_nanosec);
+
+/**
+ * @brief Set the Domain Participant builtin discovery initial announcements duration (seconds).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] period_sec Builtin discovery initial announcements duration seconds.
+ *
+ * @throw ElementInvalid Exception if the provided seconds are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_initial_announcements_period_sec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& period_sec);
+
+/**
+ * @brief Set the Domain Participant builtin discovery initial announcements duration (nanoseconds).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] period_nanosec Builtin discovery initial announcements duration nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided nanoseconds are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_initial_announcements_period_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& period_nanosec);
+
+/**
+ * @brief Set the Domain Participant builtin discovery client announcement period.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] period_sec Builtin discovery client announcement period seconds.
+ * @param[in] period_nanosec Builtin discovery client announcement period nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided client announcement period arguments are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_client_announcement_period(
```

Comment:
This `set` function with both `sec` and `nanosec` arguments is no longer required, we changed it into two individual set functions for `sec` and `nanosec`